### PR TITLE
Documentation updates for 10.4

### PR DESCRIPTION
**Pull request by IndyIndyIndy**
- Removed mentions of GMENU* and IMGMENU* everywhere
- Changed/simplified several broken/outdated typoscript examples
- Updated EDITPANEL (with mention to external fe-editing extensions)
- Removed many typoscript options, that are no longer present in 10.4

#317

### CODE REVIEW

File: Documentation/ContentObjects/Hmenu/Index.rst [hunk 1]
```diff
@@ -1605,50 +1577,62 @@ generated from this array, which was returned from the function
...
-                   '_OVERRIDE_HREF' => 'index.php?id=10&cmd=thankyou',
-               ),
-           ),
-       ),
-       array(
-           'title' => 'Products',
-           '_OVERRIDE_HREF' => 'index.php?id=14',
-       )
-     );
+   <?php
```
Comment (liayn) [inline Documentation/ContentObjects/Hmenu/Index.rst:1580]: nitpick: add a declare_strict=1

File: Documentation/ContentObjects/Hmenu/Index.rst [hunk 2]
```diff
@@ -1605,50 +1577,62 @@ generated from this array, which was returned from the function
...
+
+   namespace Vendor\MyExtension\Userfuncs;
+
+   class CustomMenu
+   {
+
+       public function makeMenuArray($content, $conf)
```
Comment (liayn) [inline Documentation/ContentObjects/Hmenu/Index.rst:1587]: maybe typehints for the params?

File: Documentation/MenuObjects/Tmenu/Index.rst [hunk 3]
```diff
@@ -95,7 +95,16 @@ TMENU
    Description
          Wraps the whole block of sub items.
 
-         **Example:** see GMENU.stdWrap
+         **Example:** ::
+
+         2 = TMENU
+         2 {
+           stdWrap.dataWrap = <ul class="{register :
+              parentProperty}"> | </ul>
```
Comment (liayn) [inline Documentation/MenuObjects/Tmenu/Index.rst:103]: this line break is strange to read

**Comment by sypets** [comment]
Backported to 10.4

**Pull request merged**